crates/libs/registry/src/key.rs

```diff
@@ -193,12 +193,24 @@ impl Key {
     /// # Safety
     ///
     /// The `PCWSTR` pointer needs to be valid for reads up until and including the next `\0`.
+    #[track_caller]
     pub unsafe fn raw_set_bytes<N: AsRef<PCWSTR>>(
         &self,
         name: N,
         ty: Type,
         value: &[u8],
     ) -> Result<()> {
+        if cfg!(debug_assertions) {
+            // RegSetValueExW expects string data to be null terminated.
+            if matches!(ty, Type::String | Type::ExpandString | Type::MultiString) {
+                debug_assert!(
+                    value.get(value.len() - 2) == Some(&0),
+                    "`value` isn't null-terminated"
+                );
+                debug_assert!(value.last() == Some(&0), "`value` isn't null-terminated");
+            }
+        }
+
```

Comment:
Nit: This doesn't quite cover the `REG_MULTI_SZ` scenario, in which each item is terminated with a `\0`. So someone may pass in an non-terminated sequence and pass these checks.

Reply:
Sure, this is just a basic smoke test for basic API conformance of strings for internal validation. Multi-strings are validated elsewhere.

Reply:
In other words, this is still a raw `unsafe` function but at least now it provides the most basic string debug assertion.

Reply:
Understood. Could also just throw in something like

```rust
if matches!(ty, Type::MultiString) {
    debug_assert!(
        value.get(value.len() - 2..value.len()) == Some(&[0, 0]),
        "`value` sequence isn't null-terminated"
    );
}
```

Not required, will mark as resolved.